testing/skia_gold_client/README.md

```diff
@@ -1,70 +1,111 @@
+import 'package:skia_gold_client/skia_gold_client.dart';
 
-```yaml
-            "dependencies": [
-                {
-                    "dependency": "goldctl",
-                    "version": "git_revision:<sha>"
-                }
-            ],
+void main() async {
+  // Create a temporary working directory.
+  final io.Directory tmpDirectory = io.Directory.systemTemp.createTempSync('skia_gold_wd');
+  try {
+    final SkiaGoldClient client = SkiaGoldClient(tmpDirectory);
+    await client.auth();
+    // ...
+  } finally {
+    tmpDirectory.deleteSync(recursive: true);
+  }
+}
 ```
 
-3. Add dependency in `pubspec.yaml`:
+Once you have an authorized instance, use `addImg` to upload a screenshot:
 
-```yaml
-dependencies:
-  # needed for skia_gold_client to avoid a cache miss.
-  engine_repo_tools:
-    path: <relative-path>/tools/pkg/engine_repo_tools
-  skia_gold_client:
-    path: <relative-path>/testing/skia_gold_client
+```dart
+await client.addImg(
+  'my-screenshot',
+  io.File('path/to/screenshot.png'),
+  screenshotSize: 400, // i.e. a 20x20 image
+);
 ```
 
-4. Use the client:
+## Configuring CI
 
-```dart
-import 'package:skia_gold_client/skia_gold_client.dart';
+Currently[^1], the client is only available on Flutter Engine's CI platform, and
+will fail to authenticate if run elsewhere.
 
-Future<void> main() {
-    final Directory tmpDirectory = Directory.current.createTempSync('skia_gold_wd');
-    final SkiaGoldClient client = SkiaGoldClient(
-        tmpDirectory,
-        dimensions: <String, String> {'<attribute-name>': '<attribute-value>'},
-    );
-
-    try {
-        if (SkiaGoldClient.isAvailable()) {
-            await client.auth();
-
-            await client.addImg(
-                '<file-name>',
-                File('gold-file.png'),
-                screenshotSize: 1024,
-            );
-        }
-    } catch (error) {
-        // Failed to authenticate or compare pixels.
-        stderr.write(error.toString());
-        rethrow;
-    } finally {
-        tmpDirectory.deleteSync(recursive: true);
-    }
-}
-```
+To use the client in CI, you'll need to make two changes:
+
+[^1]:
+    The `flutter/flutter` repository has a workaround which downloads digests
+    and does basic local image comparison, but because we have forked the
+    client and not kept it up-to-date, we cannot use that workaround. Send
+    a PR or file an issue if you'd like to see this fixed!
+
+1. **Add a dependency on `goldctl`**
+
+   In your task's configuration in [`.ci.yaml`](../../.ci.yaml) file, add a
+   dependency on `goldctl`:
+
+   ```diff
+   # This is just an example.
+   targets:
+     - name: Linux linux_android_emulator_tests
+       properties:
+         config_name: linux_android_emulator
+         + dependencies: >-
+         +   [
+         +     {"dependency": "goldctl", "version": "git_revision:720a542f6fe4f92922c3b8f0fdcc4d2ac6bb83cd"}
+         +   ]
```

Comment:
Does this render correctly?  I would have expected the pluses and minuses at the beginning of the line

Reply:
Hmm good point, let me try again